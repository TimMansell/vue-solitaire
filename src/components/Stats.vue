<template>
  <div class="stats">Games played: {{ userStats.totalGames }}</div>
</template>

<script>
import { mapGetters } from 'vuex';

export default {
  name: 'Stats',
  computed: {
    ...mapGetters('dbModule', ['userStats']),
  },
};
</script>

<style scoped lang="scss">
.stats {
  margin-bottom: var(--mg-sm);
  color: var(--text-primary);
  font-size: var(--font-size);

  @media (min-width: $bp-md) {
    margin-bottom: var(--mg-md);
    font-size: var(--font-size-lg);
  }
}
</style>
