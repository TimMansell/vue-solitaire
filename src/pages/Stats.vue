<template>
  <GameOverlay center-content show-close data-test="stats">
    <template #title> Your Stats </template>
    <template #msg>
      <Stats />
    </template>
  </GameOverlay>
</template>

<script>
import GameOverlay from '@/components/GameOverlay.vue';
import Stats from '@/components/Stats.vue';

export default {
  name: 'StatsPage',
  components: {
    GameOverlay,
    Stats,
  },
};
</script>
