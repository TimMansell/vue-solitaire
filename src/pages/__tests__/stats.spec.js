import { shallowMount } from '@vue/test-utils';
import Stats from '@/pages/Stats.vue';

describe('Stats.vue', () => {
  it('renders the component without crashing', () => {
    const wrapper = shallowMount(Stats);

    expect(wrapper.isVisible()).toBe(true);
  });
});
