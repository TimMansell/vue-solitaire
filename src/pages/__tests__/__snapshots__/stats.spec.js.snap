// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Stats.vue matches snapshot 1`] = `
<gameoverlay-stub centercontent="true" showclose="true" data-test="stats"> Your Stats <stats-stub></stats-stub>
</gameoverlay-stub>
`;
